src/modules/sidebar/partials/TransferForm.vue

```diff
@@ -0,0 +1,187 @@
+<template>
+	<div class="row space-B">
+		<h3>{{ t('tables', 'Share with accounts') }}</h3>
+		<NcSelect id="transfer-ownership-select" style="width: 100%;" :loading="loading" :options="options"
+			:placeholder="t('tables', 'User…')" :searchable="true" label="displayName" @search="asyncFind">
+			<template #no-options>
+				{{ t('tables', 'No recommendations. Start typing.') }}
+			</template>
+			<template #noResult>
+				{{ noResultText }}
+			</template>
+		</NcSelect>
+	</div>
+</template>
+
+<script>
+import { generateOcsUrl } from '@nextcloud/router'
+import { getCurrentUser } from '@nextcloud/auth'
+import axios from '@nextcloud/axios'
+import debounce from 'debounce'
+import { NcSelect } from '@nextcloud/vue'
+import { mapState } from 'vuex'
+import formatting from '../../../shared/mixins/formatting.js'
+import ShareTypes from '../mixins/shareTypesMixin.js'
+
+export default {
+	name: 'TransferForm',
+	components: {
+		NcSelect,
+	},
+
+	mixins: [ShareTypes, formatting],
+
+	props: {
+		newOwnerUserId: {
+			type: String,
+			default: '',
+		},
+	},
+
+	data() {
+		return {
+			query: '',
+			loading: false,
+			minSearchStringLength: 1,
+			maxAutocompleteResults: 20,
+			recommendations: [],
+			suggestions: [],
+		}
+	},
+
+	computed: {
+		...mapState(['tables', 'tablesLoading']),
+
+		isValidQuery() {
+			return this.query && this.query.trim() !== '' && this.query.length > this.minSearchStringLength
+		},
+
+		options() {
+			if (this.isValidQuery) {
+				return this.suggestions
+			}
+			return this.recommendations
+		},
+
+		noResultText() {
+			if (this.loading) {
+				return t('tables', 'Searching …')
+			}
+			return t('tables', 'No elements found.')
+		},
+	},
+
+	mounted() {
+		this.getRecommendations()
+	},
+
+	methods: {
+		addShare(id) {
+			this.newOwnerUserId = id
+		},
+		async asyncFind(query) {
+			this.query = query.trim()
+			if (this.isValidQuery) {
+				this.loading = true
+				await this.debounceGetSuggestions(query)
+			}
+		},
+
+		async getSuggestions(search) {
+			this.loading = true
+
+			const shareType = [
+				this.SHARE_TYPES.SHARE_TYPE_USER,
+			]
+
+			const request = await axios.get(generateOcsUrl('apps/files_sharing/api/v1/sharees'), {
+				params: {
+					format: 'json',
+					itemType: 'file',
+					search,
+					perPage: this.maxAutocompleteResults,
+					shareType,
+				},
+			})
```

Comment:
When trying to find users to transfer ownership to, we need to load that from somewhere? I just copied this from the "sharing" functionality. Not sure if we need another api endpoint for this or reusing this one works